pkg/controller/util/node/controller_utils.go

```diff
@@ -123,50 +122,35 @@ func SetPodTerminationReason(kubeClient clientset.Interface, pod *v1.Pod, nodeNa
 	return updatedPod, nil
 }
 
-func MarkAllPodsNotReadyInPartitions(kubeClients []clientset.Interface, node *v1.Node) error {
-	errString := ""
-	for i, kubeClient := range kubeClients {
-		err := MarkAllPodsNotReady(kubeClient, node)
-		if err != nil {
-			errString += fmt.Sprintf("Error in client #%d: %v", i, err)
-		}
-	}
-
-	if len(errString) != 0 {
-		return fmt.Errorf(errString)
-	}
-
-	return nil
-}
-
-// MarkAllPodsNotReady updates ready status of all pods running on
+// MarkPodsNotReady updates ready status of given pods running on
 // given node from master return true if success
-func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
-	nodeName := node.Name
+func MarkPodsNotReady(kubeClient clientset.Interface, pods []*v1.Pod, nodeName string) error {
 	klog.V(2).Infof("Update ready status of pods on node [%v]", nodeName)
-	opts := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector(api.PodHostField, nodeName).String()}
-	pods, err := kubeClient.CoreV1().PodsWithMultiTenancy(metav1.NamespaceAll, metav1.TenantAll).List(opts)
-	if err != nil {
-		return err
-	}
 
 	errMsg := []string{}
-	for _, pod := range pods.Items {
+	for i := range pods {
 		// Defensive check, also needed for tests.
-		if pod.Spec.NodeName != nodeName {
+		if pods[i].Spec.NodeName != nodeName {
 			continue
 		}
 
+		// Pod will be modified, so making copy is requiered.
+		pod := pods[i].DeepCopy()
 		for _, cond := range pod.Status.Conditions {
 			if cond.Type == v1.PodReady {
 				cond.Status = v1.ConditionFalse
 				if !utilpod.UpdatePodCondition(&pod.Status, &cond) {
 					break
 				}
 				klog.V(2).Infof("Updating ready status of pod %v to false", pod.Name)
-				_, err := kubeClient.CoreV1().Pods(pod.Namespace).UpdateStatus(&pod)
+				_, err := kubeClient.CoreV1().Pods(pod.Namespace).UpdateStatus(pod)
```

Comment:
Shouldn't this use multitenancy API?

Reply:
only GET operation has the tenancy-based APIs. write OPs ( update,delete ) has the pod itself as parameter so it will be unique anyways. the change here is the pod pointer, function wise, it is not changed.